lib/filesystem.nix

```diff
@@ -54,4 +54,24 @@
       dir + "/${name}"
   ) (builtins.readDir dir));
 
+  # find: String -> Path -> [ Path ]
+  #
+  # Given a suffix and directory, return a filtered list containing the full
+  # path to all files matching the given suffix recursively.
+  #
+  # Example:
+  #   find ".csv" "${glibc.dev}/include/protocols"
+  #
+  # Returns:
+  #  [
+  #   "/nix/store/p5vg4rdba95snvxd7p0km695jnm8vvwh-glibc-2.35-163-dev/include/protocols/routed.h"
+  #   "/nix/store/p5vg4rdba95snvxd7p0km695jnm8vvwh-glibc-2.35-163-dev/include/protocols/rwhod.h"
+  #   "/nix/store/p5vg4rdba95snvxd7p0km695jnm8vvwh-glibc-2.35-163-dev/include/protocols/talkd.h"
+  #   "/nix/store/p5vg4rdba95snvxd7p0km695jnm8vvwh-glibc-2.35-163-dev/include/protocols/timed.h"
+  # ]
+  find = suffix: dir: (builtins.filter
```

Comment:
I find it a bit too restrictive to search by suffix only, with a function named as generically as `find`, maybe rename to `findBySuffix` ?

Reply:
We're pretty limited to making this ergonomic though. With `builtins.match` we would force everyone to have to write regex. Then we can't have it be as easy as a shell `find ./. -name '*.h'` would be.

Either way, I've renamed it and added a more generic `find` function which is the same but operating on a `builtins.match` that you may like.